pkg/validator/promql_expression_helpers.go

```diff
@@ -34,6 +35,63 @@ func getExpressionUsedLabels(expr string) ([]string, error) {
 	return slices.Compact(usedLabels), nil
 }
 
+// Returns true in case metric name selector matches given regexp and a list of other labels beside __name__, false and empty list otherwise.
+func labelsUsedInSelectorForMetricRegexp(selector *parser.VectorSelector, metricRegexp string) (usedLabels []string, metricUsed bool) {
```

Comment:
Compiling is out of scope of the function and if not, the error should be returned
```suggestion
func labelsUsedInSelectorForMetricRegexp(selector *parser.VectorSelector, metricRegexp *regexp.Regexp) (usedLabels []string, metricUsed bool) {
```

Reply:
https://github.com/FUSAKLA/promruval/pull/70#discussion_r1712989880